src/hooks/useTimeRemainingInBatch.ts

```diff
@@ -13,3 +14,70 @@ export function useTimeRemainingInBatch(): number {
 
   return timeRemaining
 }
+
+interface SecondsRemainingResult {
+  secondsRemaining: number
+  checkTime: boolean
+}
+
+const checkIfTime = (seconds: number): SecondsRemainingResult => {
+  const secondsRemaining = getSecondsRemainingInBatch()
+  return {
+    secondsRemaining,
+    checkTime: secondsRemaining < seconds,
+  }
+}
+
+export function useCheckWhenTimeRemainingInBatch(
+  seconds: number,
+  callback: (SResult: SecondsRemainingResult) => void,
+): void {
+  // allow for fluid callback change without resetting interval
+  const callbackRef = useRef(callback)
+  callbackRef.current = callback
+
+  useEffect(() => {
+    // how often to check
+    // when there are ${seconds} seconds left in the batch
+    const checkIntervalSeconds = BATCH_TIME - seconds
+    if (checkIntervalSeconds <= 0) {
+      console.warn(`seconds given ${seconds} > BATCH_TIME ${BATCH_TIME}`)
+      return
+    }
+
+    const currentCheck = checkIfTime(seconds)
+    let id: number
+
+    if (currentCheck.checkTime) {
+      // setCheckResult(currentCheck)
+      callbackRef.current(currentCheck)
+    }
+
+    // const nextCheckInSeconds = currentCheck.secondsRemaining + checkIntervalSeconds
```

Comment:
ubernit: It looks super effective ;) 
I was expecting sth more like check evey X secons for every batch. Your solution is much fancier :) 

Anyways, I would add an example in the comments, to make it more clear what you try to do. If the check is fired at minute 4:05, and if it's fired at minute 2:10 (i.e. at boot time)